bundles/com.espressif.idf.core/src/com/espressif/idf/core/tools/EimLoader.java

```diff
@@ -0,0 +1,507 @@
+		{
+			String line;
+			while ((line = reader.readLine()) != null)
+			{
+				if (line.contains("/Volumes/")) //$NON-NLS-1$
+				{
+					String[] parts = line.split("\t"); //$NON-NLS-1$
+					for (String part : parts)
+					{
+						if (part.startsWith("/Volumes/")) //$NON-NLS-1$
+						{
+							volumePath = part.trim();
+							break;
+						}
+					}
+				}
+			}
+		}
+
+		if (volumePath == null)
+			throw new IOException("Failed to mount DMG: Volume path not found."); //$NON-NLS-1$
+
+		File[] apps = new File(volumePath).listFiles((dir, name) -> name.endsWith(".app")); //$NON-NLS-1$
+		if (apps == null || apps.length == 0)
+			throw new FileNotFoundException("No .app found inside DMG."); //$NON-NLS-1$
+
+		File appBundle = apps[0];
+		Path targetAppPath = Paths.get("/Applications", appBundle.getName()); //$NON-NLS-1$
+
+		logMessage("Copying app to /Applications...\n"); //$NON-NLS-1$
+
+		// Copy to /Applications
+		ProcessBuilder copyBuilder = new ProcessBuilder("cp", "-R", appBundle.getAbsolutePath(), //$NON-NLS-1$ //$NON-NLS-2$
+				targetAppPath.toString());
+		copyBuilder.inheritIO().start().waitFor();
+
+		logMessage("Unmounting DMG...\n"); //$NON-NLS-1$
+		new ProcessBuilder("hdiutil", "detach", volumePath).start().waitFor(); //$NON-NLS-1$ //$NON-NLS-2$
+
+		String eimPath = targetAppPath.toString().concat("/Contents/MacOS/eim"); //$NON-NLS-1$
+		new IDFEnvironmentVariables().addEnvVariable(IDFEnvironmentVariables.EIM_PATH, eimPath);
+		return eimPath;
+	}
+
+
+	public void downloadEim(IProgressMonitor monitor)
+	{
+		try
+		{
+			monitor.beginTask("Downloading EIM GUI...", IProgressMonitor.UNKNOWN); //$NON-NLS-1$
+			JsonObject root = fetchJson();
+			JsonArray assets = root.getAsJsonArray("assets"); //$NON-NLS-1$
+			Optional<JsonObject> match = findMatchingAsset(assets);
+
+			if (match.isEmpty())
+			{
+				listener.onError("No suitable EIM GUI asset found.", null); //$NON-NLS-1$
+				monitor.done();
+				return;
+			}
+
+			JsonObject asset = match.get();
+			String name = asset.get("name").getAsString(); //$NON-NLS-1$
+			String downloadUrl = asset.get("browser_download_url").getAsString(); //$NON-NLS-1$
+
+			Files.createDirectories(DOWNLOAD_DIR);
+			Path downloadPath = DOWNLOAD_DIR.resolve(name);
+
+			downloadFile(downloadUrl, downloadPath, listener, monitor);
+			Path eimPath = Paths.get(EimConstants.USER_EIM_DIR);
+			Files.createDirectories(eimPath);
+			
+			if (name.endsWith(".zip")) //$NON-NLS-1$
+			{
+				Path extracted = unzip(downloadPath, eimPath);
+				listener.onCompleted(extracted.toAbsolutePath().toString());
+			}
+			else if (name.endsWith(".exe")) //$NON-NLS-1$
+			{
+				eimPath = Paths.get(eimPath.toString().concat("\\").concat(name)); //$NON-NLS-1$
+				Files.copy(downloadPath, eimPath, StandardCopyOption.REPLACE_EXISTING);
+				listener.onCompleted(eimPath.toString());
+			}
+			else 
+			{
+				listener.onCompleted(downloadPath.toAbsolutePath().toString());
+			}
+		}
+		catch (Exception e)
```

Comment:
It’s common practice to catch more specific exceptions to make debugging easier.

Reply:
Done